qutebrowser/completion/models/sortfilter.py

```diff
@@ -140,21 +133,33 @@ def filterAcceptsRow(self, row, parent):
         if parent == QModelIndex() or not self.pattern:
             return True
 
-        try:
-            return self.srcmodel.custom_filter(self.pattern, row, parent)
-        except NotImplementedError:
-            for col in self.srcmodel.columns_to_filter:
-                idx = self.srcmodel.index(row, col, parent)
-                if not idx.isValid():
-                    # No entries in parent model
-                    continue
-                data = self.srcmodel.data(idx)
-                if not data:
-                    continue
-                elif self.pattern_re.search(data):
-                    return True
+        # first check the cache
```

Comment:
Ok, there's one tiny little bit I forgot. For reproducibility you either have to have a lot of rows (i.e. no `--temp-basedir`) or add an artificial delay before the message in `filterAcceptsRow`.

The filtering is probably triggered by the `invalidate()` call as you say, but there are no markers for benchmarking.